src/Symfony/Component/Cache/Marshaller/IdentityMarshaller.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Marshaller;
+
+/**
+ * @author Ahmed TAILOULOUTE <ahmed.tailouloute@gmail.com>
+ */
+class IdentityMarshaller implements MarshallerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function marshall(array $values, ?array &$failed): array
+    {
+        foreach ($values as $key => $value) {
+            if (!\is_string($value)) {
+                throw new \RuntimeException(sprintf('%s accepts only string as data.', __METHOD__));
```

Comment:
LogicException from the component instead

Reply:
updated